hs_core/templates/search/search.html

```diff
@@ -333,55 +530,136 @@ <h4 class="panel-title">
-                    existed = true;
-                }
+        var createBoundsForMarkers = function(markers) {
+            var bounds = new google.maps.LatLngBounds();
+            $.each(markers, function() {
+                bounds.extend(this.getPosition());
             });
-            if (!existed){
-                test.info_link = '"<a href="' + test.get_absolute_url + '">' + test.title + '</a>';
-                modified_points_data.push(test);
+            return bounds;
+        }
+
+        var getBoundsZoomLevel = function (bounds, mapDim) {
+            var WORLD_DIM = {height: 256, width: 256};
+            var ZOOM_MAX = 21;
+
+            function latRad(lat) {
+                var sin = Math.sin(lat * Math.PI / 180);
+                var radX2 = Math.log((1 + sin) / (1 - sin)) / 2;
+                return Math.max(Math.min(radX2, Math.PI), -Math.PI) / 2;
             }
+
+            function zoom(mapPx, worldPx, fraction) {
+                return Math.floor(Math.log(mapPx / worldPx / fraction) / Math.LN2);
+            }
+
+            var ne = bounds.getNorthEast();
+            var sw = bounds.getSouthWest();
+
+            var latFraction = (latRad(ne.lat()) - latRad(sw.lat())) / Math.PI;
+
+            var lngDiff = ne.lng() - sw.lng();
+            var lngFraction = ((lngDiff < 0) ? (lngDiff + 360) : lngDiff) / 360;
+
+            var latZoom = zoom(mapDim.height, WORLD_DIM.height, latFraction);
+            var lngZoom = zoom(mapDim.width, WORLD_DIM.width, lngFraction);
+
+            return Math.min(latZoom, lngZoom, ZOOM_MAX);
         }
 
-        var checkDuplicateBoxResults = function (modified_boxes_data, test){
-            var test_lat = (parseFloat(test.northlimit) + parseFloat(test.southlimit)) / 2;
-            var test_lng = (parseFloat(test.eastlimit) + parseFloat(test.westlimit)) / 2;
-            var existed = false;
-            modified_boxes_data.forEach(function(item){
-                var item_lat = (parseFloat(item.northlimit) + parseFloat(item.southlimit)) / 2;
-                var item_lng = (parseFloat(item.eastlimit) + parseFloat(item.westlimit)) / 2;
+        var generateLegend = function() {
+            var map_legend = document.getElementById('legend');
+            map.controls[google.maps.ControlPosition.RIGHT_TOP].push(map_legend);
+            map_legend.style.background = "white";
+            var legend_content = [];
+            legend_content.push("<p style='margin-left:5px'><img src='//cdn.rawgit.com/googlemaps/js-marker-clusterer/gh-pages/images/m2.png'> number of sites </p>");
+            legend_content.push("<p style='margin-left:20px'><img src='//cdn.rawgit.com/Concept211/Google-Maps-Markers/master/images/marker_redN.png'> N = number of point coverage resources</p>");
+            legend_content.push("<p style='margin-left:20px'><img src='//cdn.rawgit.com/Concept211/Google-Maps-Markers/master/images/marker_blueN.png'> N = number of box coverage resources</p>");
```

Comment:
Same comments as above, it might be better to put these icon images on our server rather than depending on an external server up to retrieve it on the fly